lib/api/v3/work_packages/form/work_package_schema_representer.rb

```diff
@@ -91,6 +91,19 @@ class WorkPackageSchemaRepresenter < ::API::Decorators::Single
                                                           current_user: current_user)
                    }
 
+          property :priority,
+                   exec_context: :decorator,
+                   getter: -> (*) {
+                     priority_origin = represented
+
+                     if represented.persisted? && represented.priority_id_changed?
+                       priority_origin = represented.class.find(represented.id)
+                     end
```

Comment:
@NobodysNightmare, @Azure7111 

This logic, getting the priorities that are assignable before the change in priority, is probably not needed. There is no change in assignable_priorities when the priority is changed.

Having read through the code I wrote above, the same applies to the version although that will have to be revised once the project can be changed. But that is true for other fields as well.